src/format/utils.rs

```diff
@@ -81,6 +81,21 @@ impl<'a> SizeLimiter<'a> {
             count: 0,
         }
     }
+
+    /// Pad with the specified character.
+    pub(crate) fn pad(&mut self, c: u8, n: usize) -> Result<(), Error> {
+        if self.count + n > self.size_limit {
+            return Err(Error::FormattedStringTooLarge);
+        }
+
+        for _ in 0..n {
+            self.inner.write_all(&[c])?;
```

Comment:
Here we write 1 byte at a time, but this was already the case before when using `write_fmt()`:
https://github.com/rust-lang/rust/blob/a4a11aca5ecf24dfff3c00715641026809951305/library/core/src/fmt/mod.rs#L1756

Reply:
we can probably improve this a bit by using combination of `iter::repeat([c, c, c, ...])` and take, right?